packit_service/worker/github_handlers.py

```diff
@@ -517,3 +460,130 @@ def run(self) -> HandlerResults:
                 )
 
         return HandlerResults(success=True, details={})
+
+
+@add_to_pr_comment_mapping
+class GitHubPullRequestCommentCoprBuildHandler(PullRequestCommentHandler):
+    """ Issue handler for comment `/packit copr-build` """
+
+    name = PullRequestCommentAction.copr_build
+
+    def __init__(self, config: Config, event: PullRequestCommentEvent):
+        super(GitHubPullRequestCommentCoprBuildHandler, self).__init__(
+            config=config, event=event
+        )
+        self.config = config
+        self.event = event
+        self.project: GitProject = self.github_service.get_project(
+            repo=event.base_repo_name, namespace=event.base_repo_namespace
+        )
+        # Get the latest pull request commit
+        self.event.commit_sha = self.project.get_all_pr_commits(self.event.pr_id)[-1]
+        self.package_config: PackageConfig = get_package_config_from_repo(
+            self.project, self.event.commit_sha
+        )
+        self.package_config.upstream_project_url = event.https_url
+
+    def get_tests_for_build(self) -> Optional[JobConfig]:
+        """
+        Check if there are tests defined
+        :return: JobConfig or None
+        """
+        for job in self.package_config.jobs:
+            if job.job == JobType.tests:
+                return job
+        return None
+
+    def run(self) -> HandlerResults:
+
+        collaborators = self.project.who_can_merge_pr()
+        r = BuildStatusReporter(self.project, self.event.commit_sha)
+        if self.event.github_login not in collaborators:
+            msg = "Only collaborators can trigger Packit-as-a-Service"
+            r.set_status("failure", msg, PRCheckName.get_build_check())
```

Comment:
Here's a scenario:
1. I create a PR on a project where I don't have commit rights
2. Someone with commit access types `/packit copr-build`
3. Build passes, check is now green
4. I type `/packit copr-build`
5. Packit service sets the check to "Only collaborators can trigger Packit-as-a-Service"

Is this a correct assumption? If so, in such a case, it would be better for p-s to comment on the PR instead of setting the status.

Reply:
Yes, your assumption is right. I will add a PR comment instead of setting status.